ansible/roles/nova-flavors/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+---
+    - name: Ensure Python requirements are installed
+      pip:
+        requirements: "{{ '/'.join([role_path, 'files', 'requirements.txt']) }}"
+        extra_args: >-
+          -c {{ flavors_python_upper_constraints_url }}
+        virtualenv: "{{ flavors_virtualenv_path }}"
+
+    - name: Register Nova flavors
+      os_nova_flavor:
+        auth_type: password
+        name: "{{ item.name | default(item.resource_class) }}"
+        # FIXME(w-miller): don't necessarily assume the first disk?
+        disk: "{{ node_types[item.node_type].volumes.0.capacity | default('0')
+                  | size_string_to_gb }}"
+        ram: "{{ node_types[item.node_type].memory_mb }}"
+        vcpus: "{{ node_types[item.node_type].vcpus }}"
+        # NOTE(w-miller): I'm not quite sure whether this is janky or beautiful.
+        #
+        #   * Set hardware specs to zero here for scheduling purposes.
+        #   * Add the resource class name.
+        #   * Add required and forbidden traits.
+        #   * Add any custom specs from the user.
+        extra_specs: >-
```

Comment:
This might be easier on the eye if each of the dicts being combined were defined as a task variable

Reply:
Ok.

Reply:
Sure.